File: src/main/java/org/jukeboxmc/event/entity/EntityRemoveEffectEvent.java
package org.jukeboxmc.event.entity;

import org.jukeboxmc.entity.Entity;
import org.jukeboxmc.event.Cancellable;
import org.jukeboxmc.potion.Effect;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class EntityRemoveEffectEvent extends EntityEvent implements Cancellable {

    private final Effect effect;

    public EntityRemoveEffectEvent( Entity entity, Effect effect ) {
        super( entity );
        this.effect = effect;
    }

    public Effect getEffect() {
        return this.effect;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/entity/EntityRemoveEffectEvent.kt
package org.jukeboxmc.api.event.entity

import org.jukeboxmc.api.effect.Effect
import org.jukeboxmc.api.entity.Entity
import org.jukeboxmc.api.event.Cancellable

class EntityRemoveEffectEvent(
        entity: Entity,
        private var effect: Effect
) : EntityEvent(entity), Cancellable {

    fun getEffect(): Effect {
        return this.effect
    }
}